resultbrowser/src/js/resultbrowser.js

```diff
@@ -95,211 +183,233 @@
         // Check for presence of HAR file by simply loading it via AJAX
         // -> In case AJAX call fails, HAR file is assumed to be missing
         //    and 'View as HAR' link will be visually hidden
-        $.ajax({ url: 'data.har', dataType: 'json' }).fail(function(){
-          $('.har', $transaction).hide();
+        ajax("data.har", { dataType: 'json' }).then((response) => {
+            if (!response.ok) {
+                transaction.querySelectorAll(":scope .har").forEach(hide);
+            }
         });
 
         initEvents();
     }
 
     function initEvents() {
-        var $highlight = $('#highlightSyntax'),
-            $beautify  = $('#beautify');
+        const highlight = getElementById("highlightSyntax");
+        const beautify = getElementById("beautify");
 
-        if(extras.highlight) {
-            $highlight.click(function(){
-                $requestText.each(function(i, e) {
-                    hljs.highlightBlock(e)
-                });
-            }).removeAttr('disabled');
-        }
+        // init listeners for highlight button
+        highlight.addEventListener("click", function () {
+            hljs.highlightBlock(requestText);
+        });
 
-        if(extras.beautify.js || extras.beautify.css || extras.beautify.html) {
-            $beautify.click(function(){
-                var s = $requestText.text();
-                // CSS
-                if($requestText.hasClass('css')) {
-                    try {
-                        s = css_beautify(s);
-                    }
-                    catch(e){}
+        // init listeners for beautify button
+        beautify.addEventListener("click", function () {
```

Comment:
Attach listener if and only if beautifying is possible (extras.beautify.[css|js|html})

Reply:
Done